provider/source/src/lib.rs

```diff
@@ -187,6 +227,17 @@ impl SourceDataProvider {
         })
     }
 
+    /// TODO
```

Comment:
Nit: Please write the docs now while this is fresh. It takes 60 seconds. (In general, not just this PR)